### PR TITLE
feat(3197): Table cell component for stages in pipeline list jobs view

### PR DESCRIPTION
**Pull request by tkyi**
## Context
The new jobs UI makes use of custom components for each cell in the table.  This creates a new component for handling the stage name for the job.

## Objective
Create a glimmer component for the jobs.
V1:
Screenshot 2024-11-09 at 10 03 13 PM

V2:
Screenshot 2024-11-09 at 10 05 27 PM

## References
https://github.com/screwdriver-cd/screwdriver/issues/3197
https://github.com/screwdriver-cd/screwdriver/issues/3200

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/pipeline-list-stage-cell/component.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+
+export default Component.extend({
+  stageName: computed('record.stage.stageName', {
```
Comment (VonnyJap) [inline app/components/pipeline-list-stage-cell/component.js:5]: Locally I got an error when viewing something without a stage. 
Screenshot 2024-11-12 at 4 47 49 PM

Comment (VonnyJap) [reply app/components/pipeline-list-stage-cell/component.js:5]: Config for reference https://github.com/screwdriver-cd-test/functional-commands/blob/master/screwdriver.yaml

Comment (VonnyJap) [reply app/components/pipeline-list-stage-cell/component.js:5]: After some investigation, this issue is caused when a pipeline has jobs that never run before, the change to the jobs list view to include the stages is not able to handle this at the moment. 

To reproduce: create any new pipeline and don't run any builds, check into the jobs list view and console.

**Review by VonnyJap — APPROVED** [review]

File: app/components/pipeline-list-stage-cell/component.js (hunk 1 above)
Comment (tkyi) [reply app/components/pipeline-list-stage-cell/component.js:5]: Strangely this bug doesn't show up the first time the jobs list view loads, only after refreshing. Fixed it by putting a default value `N/A`

**Pull request merged**